src/containers/AddProgram.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component, PropTypes } from 'react';
+import { Actions } from 'react-native-router-flux';
+import { View, ListView, Text, StyleSheet } from 'react-native';
+import { Container, Content, List, ListItem, InputGroup, Input } from 'native-base';
+
+import { connect } from 'react-redux';
+import * as actions from '../actions/program';
+
+import styles from '../styles';
+
+class AddProgramContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newProgramName: ''
+    }
+    this.props.component.onRight = () => {
+      this.props.addProgram(this.props.site_id, this.state.newProgramName);
+      Actions.pop();
+    }
+    this.props.component.rightTitle = 'Done';
+  }
+
+  render() {
+    return (
+      <Container style={[styles.container, styles.grayBg]}>
+          <Content>
+            <InputGroup style={styles.inputGroup}>
+                <Input placeholder="Program Name" onChangeText={(newProgramName) => this.setState({newProgramName})}/>
```

Comment:
I did a bit of research online and it seems like keeing inputs in the local state for the component is a good idea, so that's why I threw it in there